### PR TITLE
Event service

### PR DESCRIPTION
**Pull request by carmina-c**
- Added event service for refreshing task list when different actions are performed (adding a task, updating a task, deleting a task, marking a task as complete/in progress)
- Close the dialog after "add a task" button is clicked if the action is successful
- Different background color for completed tasks

### CODE REVIEW

File: src/app/task-list/task-list.component.ts [hunk 1]
```diff
@@ -12,21 +14,29 @@ import { TaskFormComponent } from "../task-form/task-form.component";
...
+		private apiService: ApiService,
+		private eventService: EventService
 	) {}
 
 	public ngOnInit(): void {
+		this.getTasks();
+		this.handleEvents();
+	}
+
+	public getTasks(): void {
```
Comment (danielursu) [inline src/app/task-list/task-list.component.ts]:
```suggestion
	private prepareTasks(): void {
```

File: src/app/task-list/task-list.component.ts [hunk 2]
```diff
@@ -12,21 +14,29 @@ import { TaskFormComponent } from "../task-form/task-form.component";
...
 	public constructor(
 		private dialogService: DialogService,
-		private apiService: ApiService
+		private apiService: ApiService,
+		private eventService: EventService
 	) {}
 
 	public ngOnInit(): void {
+		this.getTasks();
+		this.handleEvents();
```
Comment (danielursu) [inline src/app/task-list/task-list.component.ts]: move to the constructor

File: src/app/task/task.component.ts [hunk 3]
```diff
@@ -61,6 +70,7 @@ export class TaskComponent {
 				title: this.task.title,
 				description: this.task.description,
 				isCompleted: this.task.isCompleted,
+				userId: this.task.userId,
 				clickType: clickTypeParam
```
Comment (danielursu) [inline src/app/task/task.component.ts]: do we need this anymore?

**Review by danielursu — CHANGES_REQUESTED** [review]

File: src/app/task/task.component.ts (hunk 3 above)
Comment (carmina-c) [reply src/app/task/task.component.ts]: nope

File: src/app/task-list/task-list.component.ts (hunk 2 above)
Comment (danielursu) [reply src/app/task-list/task-list.component.ts]: move only the subscriptions to the constructor. Leave prepareTasks() into onInit

File: src/app/task/task.component.ts (hunk 3 above)
Comment (danielursu) [reply src/app/task/task.component.ts]: I was referring to clickType

**Review by danielursu — APPROVED** [review]

File: src/app/task/task.component.ts (hunk 3 above)
Comment (carmina-c) [reply src/app/task/task.component.ts]: it's used in onSubmit to call either addTask() or EditTask()

**Pull request merged**